File: my-solutions/0049-group-anagrams/solution.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """

        def getKey(s):
            return "".join(sorted(s))

        res = []
        str_map = {}
        for i in range(len(strs)):
            key = getKey(strs[i])
            if key in str_map:
                str_map[key].append(strs[i])
            else:
                str_map[key] = [strs[i]]

        # print(str_map)

        
        for key in str_map:
            res.append(str_map[key])

        return res

File: my-solutions/0049-group-anagrams/solution.cpp
class Solution {
public:
    string getKey(string s) {
        string key = "00000000000000000000000000";
        for(int i = 0; i < s.length(); i++) {
            key[(int(s[i]) - int('a'))]++;
        }
        return key;
    }
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> res;
        unordered_map<string, vector<string>> string_map;
        for(int i = 0; i < size(strs); i++) {
            string key = getKey(strs[i]);
            if(string_map.find(key) == string_map.end()) {
                string_map[key] = vector<string> {strs[i]};
            } else {
                string_map[key].push_back(strs[i]);
            }
        }

        for(const auto &i: string_map) {
            res.push_back(i.second);
        }

        return res;
    }
};
